### PR TITLE
test images: uses nanoserver

### LINKED ISSUE #77268: Kubernetes test images should include Windows images in their manifest lists
**What would you like to be added**:

/sig windows
/sig testing

/area conformance

The images used in Kubernetes tests should also include Windows images in their manifest lists. Right now, in order to run E2E tests with Windows nodes, the env var ``KUBE_TEST_REPO_LIST`` needs to be set, containing alternative repositories which contain Windows images (the ``e2eteam`` repository is mainly used). The Windows images are currently maintained and built from ``kubernetes-sigs/windows-testing``: https://github.com/kubernetes-sigs/windows-testing/blob/master/images/BuildImageManifestLists.ps1

The build script in ``kubernetes/test/images`` does not build Windows images at the moment.

**Why is this needed**:

Running E2E tests on Windows nodes will be made easier, with the official Kubernetes test images.

Secondly, Hybrid OS tests will be able to be run.

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig testing
/sig windows

**What this PR does / why we need it**:

Using Windows nanoserver container images as a base instead of the current Windows servercore image will reduce the image size by about ~10x.

However, the nanoserver image lacks several things we need:
- netapi32.dll
- powershell
- certain powershell commands
- chocolatey cannot be used

When building the nanoserver images, we are going to use a Windows servercore helper, in which we are going to install the necessary dependencies, and then copy them over to our nanoserver image, including necessary DLLs.

Other notable changes include:
- switch from wget to curl (wget was a powershell alias).
- implement in code getting the DNS suffix list and DNS server list.
- reimplement getting file permissions for mounttest.

**Which issue(s) this PR fixes**:

Partially Fixes #77268

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by kbruner** [comment]
Hello from the bug triage team! This issue has not been updated for a long time, so I'd like to check what's the status. The code freeze is starting June 25th (about 4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.19, is it still planned for this release?

**Comment by kbruner** [comment]
@claudiubelu @kubernetes/sig-windows-feature-requests @kubernetes/sig-testing Is this PR still targeted to v1.19?

**Comment by marosset** [comment]
Yes, sig-windows would like this to land in 1.19

**Comment by kbruner** [comment]
Just a reminder that code freeze for 1.19 is July 9. Is this PR still targeted for that milestone?
@kubernetes/sig-windows-bugs @claudiubelu

**Comment by kbruner** [comment]
@claudiubelu @kubernetes/sig-windows-bugs Code thaw starts tomorrow. This PR should either be ready for merging, assuming it has high enough priority (priority is not set) or moved to the 1.20 milestone.

**Comment by liggitt** [comment]
since this hasn't been reviewed/approved and we're at the end of 1.19 code freeze, moving to 1.20

/milestone v1.20

**Comment by michmike** [comment]
should reduce the time required for tests that we want to introduce with https://github.com/kubernetes/kubernetes/issues/93276

File: test/images/agnhost/Dockerfile_windows [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 ARG BASEIMAGE
+FROM claudiubelu/windows-image-builder-helper:1.0 as helper
```
Comment (dims) [inline test/images/agnhost/Dockerfile_windows:16]: is there a non-personal location for this helper image?

File: test/images/agnhost/dns/dns_windows.go [hunk 2]
```diff
@@ -20,26 +20,95 @@ import (
 	"bytes"
 	"os/exec"
 	"strings"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	//"golang.org/x/sys/windows/registry"
```
Comment (dims) [inline test/images/agnhost/dns/dns_windows.go:27]: remove this?

File: test/images/agnhost/dns/dns_windows.go [hunk 3]
```diff
@@ -20,26 +20,95 @@ import (
...
+		uintptr(unsafe.Pointer(&size)),
+	)
+
+	info := (*FixedInfo)(unsafe.Pointer(&buffer[0]))
+	return info
+}
 
 func getDNSSuffixList() []string {
-	output := runCommand("powershell", "-Command", "(Get-DnsClient)[0].SuffixSearchList")
+	/*
```
Comment (dims) [inline test/images/agnhost/dns/dns_windows.go:79]: remove commented out code?

**Comment by justaugustus** [comment]
Needs rebase

**Comment by dims** [comment]
@mkumatag @listx please review test/images/ directory then we can hand this off to @liggitt for root approval

/assign @mkumatag @listx

**Comment by spiffxp** [comment]
/cc @spiffxp @BenTheElder 
discussed at wg-k8s-infra today

File: test/images/agnhost/dns/dns_windows.go [hunk 4]
```diff
@@ -17,40 +17,131 @@ limitations under the License.
 package dns
 
 import (
-	"bytes"
-	"os/exec"
+	"fmt"
```
Comment (spiffxp) [inline test/images/agnhost/dns/dns_windows.go:20]: line unrelated

Is there any particular reason we don't have `// +build windows` as line 1 of this file?

**Review by spiffxp — COMMENTED** [review]
I can't speak to the windows specifics; this looks reasonable however I have one nit

File: test/images/agnhost/dns/dns_windows.go (hunk 4 above)
Comment (claudiubelu) [reply test/images/agnhost/dns/dns_windows.go:20]: Sure, it felt redundant, we were already relying on the file name suffix for that (conditional compilation can also be done through filename suffix). Added that line now.

**Comment by spiffxp** [comment]
needs rebase

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by spiffxp** [comment]
/assign @liggitt @thockin 
I don't have /approve rights for the `/go.sum` change

File: go.sum [hunk 5]
```diff
@@ -265,6 +265,8 @@ github.com/hashicorp/golang-lru v0.5.1 h1:0hERBMJE1eitiLkihrMvRVBYAkpHzc/J3QdDN+
 github.com/hashicorp/golang-lru v0.5.1/go.mod h1:/m3WP610KZHVQ1SGc6re/UDhFvYD7pJ4Ao+sR/qLZy8=
 github.com/hashicorp/hcl v1.0.0 h1:0Anlzjpi4vEasTeNFn2mLJgTSwt0+6sfsiTG8qcWGx4=
 github.com/hashicorp/hcl v1.0.0/go.mod h1:E5yfLk+7swimpb2L/Alb/PJmXilQ/rhwaUYs4T20WEQ=
+github.com/hectane/go-acl v0.0.0-20190604041725-da78bae5fc95 h1:S4qyfL2sEm5Budr4KVMyEniCy+PbS55651I/a+Kn/NQ=
+github.com/hectane/go-acl v0.0.0-20190604041725-da78bae5fc95/go.mod h1:QiyDdbZLaJ/mZP4Zwc9g2QsfaEA4o7XvvgZegSci5/E=
```
[the remark this answers is not in the document]
Comment (claudiubelu) [reply go.sum]: This is probably a remnant from a previous version I had, I was using this in agnhost, but no longer needed it afterwards. Removed.

[the remark this answers is not in the document]
Comment (liggitt) [reply go.sum]: thanks, looks like this fits within test/release approval now, will defer to those reviewers

**Comment by claudiubelu** [comment]
Can the /lgtm be reapplied?

**Pull request merged**